### ISSUE TITLE
Teacher forcing

### ISSUE THREAD

**Issue by karanchahal** [yunjey/pytorch-tutorial#100]
I tried to train the image captioning model, turns out it overfits terribly, beacuse of constant teacher forcing. How did you manage to train it ? It's giving overfit results to every new image it sees for which there is no caption.

**Comment by yunjey** [comment]
@karanchahal Do you mean that the model "you implemented" is heavily overfitting ?
I remember that the model's BLEU4 score is about 23 on coco validation dataset. This means that the model did not overfit so badly.

**Comment by karanchahal** [comment]
Okay , but it's very strange , I trained it from scratch and sampling gives terrible results . The same caption for all images . How many epochs did you train it for ?

**Comment by yunjey** [comment]
@karanchahal Please see [here](https://github.com/yunjey/pytorch-tutorial/blob/master/tutorials/03-advanced/image_captioning/train.py#L116) (5 epochs). The most possible reason for generating same captions is incorrect input normalization. Check input normalization between training and sampling phases.
